src/databricks/labs/ucx/hive_metastore/tables.py

```diff
@@ -481,3 +481,134 @@ def _describe(self, catalog: str, database: str, table: str) -> Table | None:
+            properties_list = list(self._iterator(raw_table.properties()))
+
+            if len(properties_list) == 0:
+                storage_properties = None
+            else:
+                formatted_items: list[str] = []
+                for each_property in properties_list:
+                    key = each_property._1()
+                    value = each_property._2()
+
+                    redacted_key = "*******"
+
+                    if key == "personalAccessToken" or key.lower() == "password":
+                        value = redacted_key
+                    elif value is None:
+                        value = "None"
+
+                    formatted_items.append(f"{key}={value}")
+                storage_properties = f"[{', '.join(formatted_items)}]"
+
+            partition_column_names = list(self._iterator(raw_table.partitionColumnNames()))
+            is_partitioned = len(partition_column_names) > 0
+
+            return Table(
+                catalog='hive_metastore',
+                database=database,
+                name=table,
+                object_type=raw_table.tableType().name(),
+                table_format=table_format,
+                location=location_uri,
+                view_text=view_text,
+                storage_properties=storage_properties,
+                is_partitioned=is_partitioned,
+            )
+        except Exception as e:  # pylint: disable=broad-exception-caught
+            logger.error(f"Couldn't fetch information for table {full_name} : {e}")
+            return None
+
+    def _crawl(self) -> Iterable[Table]:
+        """Crawls and lists tables within the specified catalog and database.
+
+        After performing initial scan of all tables, starts making parallel
+        DESCRIBE TABLE EXTENDED queries for every table.
+
+        Production tasks would most likely be executed through `tables.scala`
+        within `crawl_tables` task due to `spark.sharedState.externalCatalog`
+        lower-level APIs not requiring a roundtrip to storage, which is not
+        possible for Azure storage with credentials supplied through Spark
+        conf (see https://github.com/databrickslabs/ucx/issues/249).
+
+        See also https://github.com/databrickslabs/ucx/issues/247
+        """
+        tasks = []
+        catalog = "hive_metastore"
+        databases = self._all_databases()
+        for database in databases:
+            logger.info(f"Scanning {database}")
+            table_names = [partial(self._list_tables, database)]
+            for table_batch in Threads.strict('listing tables', table_names):
+                if len(table_batch) > 0:
+                    for table in table_batch:
+                        tasks.append(partial(self._describe, catalog, database, table))
+        catalog_tables, errors = Threads.gather("describing tables in ", tasks)
+        if len(errors) > 0:
+            logger.error(f"Detected {len(errors)} errors while scanning tables in ")
```

Comment:
```suggestion
            logger.warning(f"Detected {len(errors)} errors while scanning tables in ")
```
let's not fail the assessment if there are errors.